[analyze] Add Analyzer for Slack

abmussani opened the pull request:
### Description:
Implemented analyzer interface for Slack.

### Checklist:
* [x] Tests passing (`make test-community`)?
* [x] Lint passing (`make lint` this requires [golangci-lint](https://golangci-lint.run/usage/install/#local-installation))?

abasit-folio3 committed:
implement analyzer interface for slack

abasit-folio3 committed:
slack analyzer adjusted for new changes in main, unit test added

abasit-folio3 committed:
link detector with analyzer for slack

mcastorina reviewed (commented):
Everything here looks good, but we're missing the enumerated permissions / generated types.

abasit-folio3 committed:
added generated permissions for slack analyzer

abasit-folio3 committed:
generate permission fix, keep dot in permissions intact

abmussani committed:
removed scope from permission and put it metadata.

mcastorina commented on pkg/analyzer/analyzers/slack/slack.go:
```diff
@@ -12,8 +15,86 @@ import (
+	}
+	resource := analyzers.Resource{
+		Name:               info.User.User,
+		FullyQualifiedName: fullyQualifiedName,
```
Thoughts here about including the team ID in the fully qualified name?

mcastorina commented on pkg/analyzer/analyzers/slack/slack.go:
```diff
@@ -12,8 +15,86 @@ import (
+	var permissions []analyzers.Permission
+
+	for _, scope := range strings.Split(info.Scopes, ",") {
+		perms, ok := scope_mapping[scope]
```
Does `scope_mapping` use the generated permission type? I would feel better about this if it did, to make sure we're assigning permissions from the enumerated set.

abmussani committed:
[chore]
moved expected output of test in json file to neat the code.
added team id in fully qualified name of user resource.
check permissions before adding it in bindings.
